boa_parser/src/parser/expression/update.rs

```diff
@@ -72,14 +72,26 @@ where
                 cursor
                     .next(interner)?
                     .expect("Punctuator::Inc token disappeared");
-
+                
                 let target = UnaryExpression::new(self.name, self.allow_yield, self.allow_await)
                     .parse(cursor, interner)?;
-
-                if cursor.strict_mode() {
-                    if let Expression::Identifier(ident) = target {
-                        check_strict_arguments_or_eval(ident, position)?;
+                let strict = cursor.strict_mode();
+                // https://tc39.es/ecma262/#sec-update-expressions-static-semantics-early-errors
+                let simple = match &target {
+                    Expression::Identifier(_) if !strict => true,
```

Comment:
You can remove this, since the check already filters out the strict case

Reply:
Ah I'm sorry, I'm not sure if I understand this clearly. As far as I can tell, `check_strict_arguments_or_eval()`  does a simple match on the identifier's `Sym` and does not check for strict case. Also, the compiler fails a bunch of EcmaScript test cases when I remove this check.

Reply:
Ah, didn't see that the check was not what I had in mind. What I was talking about was doing something like:

```Rust
// ... match above
if strict {
    check_strict_arguments_or_eval(..args)?;
}
true
```

Reply:
Thanks for the suggestion. Removed it and wrapped the check for simple expression into an inline function to not repeat the same code in four different places.